great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1147,6 +1147,15 @@ def get_data_for_batch_identifiers(
     def _build_selectable_from_batch_spec(
         self, batch_spec: BatchSpec
     ) -> Union[Selectable, str]:
+        if (
+            batch_spec.get("query") is not None
+            and batch_spec.get("sampling_method") is not None
+        ):
+            raise ValueError(
```

Comment:
I am going to file a ticket to add sampling support to new experimental datasources. We should be able to use the same strategy/helper method I use here to add query support.